runtime/native/unsafe.cpp

```diff
@@ -52,18 +54,26 @@ Java_sun_misc_Unsafe_registerNatives(JNIEnv *env, jclass obj) {
 }
 
 jint
-Java_sun_misc_Unsafe_getInt__Ljava_lang_Object_2J(JNIEnv *env, jobject, jobject, jlong) {
-    UnsafeUnimplemented("Java_sun_misc_Unsafe_getInt__Ljava_lang_Object_2J");
+Java_sun_misc_Unsafe_getInt__Ljava_lang_Object_2J(JNIEnv *env, jobject unsafeObj, jobject o, jlong offset) {
```

Comment:
These also need to support static object accesses.
When `o` is NULL then the offset should refer to a static variable address.